src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.PrivateKey.cs

```diff
@@ -239,5 +339,223 @@ private CspParameters GetPrivateKeyCsp()
                 }
             }
         }
+
+        private unsafe ICertificatePal CopyWithPersistedCngKey(CngKey cngKey)
+        {
+            if (string.IsNullOrEmpty(cngKey.KeyName))
+            {
+                return null;
+            }
+
+            // Make a new pal from bytes.
+            CertificatePal pal = (CertificatePal)FromBlob(RawData, SafePasswordHandle.InvalidHandle, X509KeyStorageFlags.PersistKeySet);
+
+            CngProvider provider = cngKey.Provider;
+            string keyName = cngKey.KeyName;
+            bool machineKey = cngKey.IsMachineKey;
+
+            // CAPI RSA_SIGN keys won't open correctly under CNG without the key number being specified, so
+            // check to see if we can figure out what key number it needs to re-open.
+            int keySpec = GuessKeySpec(provider, keyName, machineKey, cngKey.AlgorithmGroup);
+
+            CRYPT_KEY_PROV_INFO keyProvInfo = new CRYPT_KEY_PROV_INFO();
+
+            fixed (char* keyNamePtr = cngKey.KeyName)
```

Comment:
Desktop seems to use it with the marshaller, but when the read got ported to core it moved to char*.  If I can easily switch it to safe read I will; but I don't want the read and the write to have different copies of the struct.